spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/source/SparkScan.java

```diff
@@ -175,7 +186,43 @@ public Statistics estimateStatistics() {
   protected Statistics estimateStatistics(Snapshot snapshot) {
     // its a fresh table, no data
     if (snapshot == null) {
-      return new Stats(0L, 0L);
+      return new Stats(0L, 0L, Collections.emptyMap());
+    }
+
+    boolean cboEnabled =
+        Boolean.parseBoolean(spark.conf().get(SQLConf.CBO_ENABLED().key(), "false"));
+    Map<NamedReference, ColumnStatistics> colStatsMap = Collections.emptyMap();
+    if (readConf.reportColumnStats() && cboEnabled) {
+      colStatsMap = Maps.newHashMap();
+      List<StatisticsFile> files = table.statisticsFiles();
+      if (!files.isEmpty()) {
+        List<BlobMetadata> metadataList = (files.get(0)).blobMetadata();
```

Comment:
As currently implemented, there is one `StatisticsFile` in the `List` for any given `Snapshot`, but there could be multiple files (corresponding to different `Snapshot`s). You can't just take the first file in the `List`.
I have put up a fix in https://github.com/apache/iceberg/pull/12482.